### PR TITLE
Make the netcdf-c library compile with gcc -ansi.

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
Primary fixes to get -ansi to work.
1. Convert all '//' C++ style comments to /*...*/ or to use #if 0...#endif
2. It turns out that when -ansi is specified, then a number of
   functions no longer are defined in the header -- but they are still
   in the .so file.
   The big example is strdup(). So, added code to include/ncconfig.h to define
   externs for those missing functions that occur in more than one place.
   These are enabled if !_WIN32 && __STDC__ == 1 (__STDC__ is supposed to
   be the equivalent compile time flag to -ansi). Note that this requires
   config.h (which references ncconfig.h) to be included in files where it is
   currently not included. Single uses will be only in the file that uses them.
3. Added mmap test for the MAP_ANONYMOUS flag to configure.ac. Apparently
   this is not always defined with -ansi.
4. fix some large integer constants in nc_test4/tst_atts3.c and nc_test4/tst_filterparser.c
   to avoid compiler complaints.
5. fix a double constant in nc_test4/tst_filterparser.c to avoid compiler complaints.

[Note I suspect numbers 4 and 5 will be a problem on big-endian machines, but we have no way to test]

Misc. Changes:
1. convert more instances of _MSC_VER to _WIN32.
2. added some debugging code to include/nctestserver.h
3. added comment about libdispatch/drc.c always being compiled.
4. modify parser generation in ncgen to remove unneeded files.

### CODE REVIEW

**Comment by DennisHeimbigner** [comment]
Beware: this pr causes failures under cygwin. It looks like there is some sort of
difference in the handling of floating point values when -ansi is specified.
One difference is the NC_ERANGE is not be returned. Another is that floating point
comparisons fail when they should succeed. No idea what is the root cause.

**Comment by edhartnett** [comment]
I would suggest that compiling on cygwin is far more important than compiling with -ansi.

**Comment by DennisHeimbigner** [comment]
Agreed. It is not a show-stopper. I can do the check for -ansi compliance under my
ubuntu VM. I am not sure in practice how/why anyone would be using -ansi
under cygwin anyway.

**Comment by DennisHeimbigner** [comment]
Correct. ok without -ansi, test failures with -ansi.

**Comment by WardF** [comment]
These fixes break compilation on OSX; removing from #1249 for the moment and will revisit with details.

File: libdispatch/dstring.c [hunk 1]
```diff
@@ -354,5 +354,28 @@ strlcat(char* dst, const char* src, size_t dsize)
...
-#endif /*!_MSC_VER*/
+#endif /*!_WIN32*/
 #endif /*!HAVE_STRLCAT*/
+
+/* It turns out the gcc -ansi under cygwin does
+   not define the following function(s) in <string.h>.
+*/
+#ifndef HAVE_STRDUP
+#if 0
+#if __STDC__ == 1 /*supposed to be same as -ansi flag */
```
Comment (t-b) [inline libdispatch/dstring.c]: I'm not understanding why this function is added when you do `#if 0` anyway.

**Comment by t-b** [comment]
As I was the one suggesting to use `-ansi` I'll had a look at the PR.

I think this PR goes definitly in the right direction but I think the mass of changes makes it difficult to review. How about making a PR with only the comment change from `//` -> `/* */`? This should make make the diff much smaller.

Regarding the missing `strdup` declaration, this function is only defined in POSIX and not in C itself, see http://man7.org/linux/man-pages/man3/strdup.3.html.

**Comment by DennisHeimbigner** [comment]
BTW agree about doing this in smaller chunks; my bad. I got carried
away in doing all the ansi fixes in one fell swoop.

**Comment by DennisHeimbigner** [comment]
See pr https://github.com/Unidata/netcdf-c/pull/1252

**Comment by WardF** [comment]
Still seeing the following on OSX, will try to debug this.

````
In file included from /Users/wfisher/Desktop/gitprojects/netcdf-c/build/config.h:596:
/Users/wfisher/Desktop/gitprojects/netcdf-c/include/ncconfigure.h:45:15: error: expected parameter declarator
extern size_t strlcat(char*,const char*,size_t);
              ^
/usr/include/secure/_string.h:115:47: note: expanded from macro 'strlcat'
                __builtin___strlcat_chk (dest, __VA_ARGS__, __darwin_obsz (dest))
                                                            ^
/usr/include/secure/_common.h:39:62: note: expanded from macro '__darwin_obsz'
#define __darwin_obsz(object) __builtin_object_size (object, _USE_FORTIFY_LEVEL > 1 ? 1 : 0)
                                                             ^
/usr/include/secure/_common.h:30:32: note: expanded from macro '_USE_FORTIFY_LEVEL'
#    define _USE_FORTIFY_LEVEL 2
````

**Comment by DennisHeimbigner** [comment]
Let me propose a wild idea:
In nccconfigure.h, change:
    extern size_t strlcat(char*,const char*,size_t);
to
    extern size_t strlcat(char*dst,const char*src,size_t count);
That is, add actual parameter names.

**Pull request merged**